### PR TITLE
ARROW-4640: [Python] Add docker-compose configuration to build and test the project without pandas installed

### PR DESCRIPTION
**Pull request by kszucs**
```bash
make -f Makefile.docker python-nopandas
make -f Makefile.docker run python-nopandas
```

OR

```bash
$ docker-compose build cpp
$ docker-compose build python
$ docker-compose build python-nopandas
$ docker-compose run --rm python-nopandas
```

The tests will fail until [ARROW-4794: [Python] Make pandas an optional test dependency](https://issues.apache.org/jira/browse/ARROW-4794) is implemented.

### CODE REVIEW

**Comment by wesm** [comment]
Thanks -- I'll take care of the pandas hard dependency issue and then return to this

File: python/Dockerfile.nopandas [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ARG PYTHON_VERSION=3.6
```
Comment (xhochy) [inline python/Dockerfile.nopandas]: Please add a license header: `NOT APPROVED: python/Dockerfile.nopandas (apache-arrow/python/Dockerfile.nopandas): false`

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by wesm** [comment]
I hope this passes now =)

**Comment by kszucs** [comment]
I'm running it.

**Comment by kszucs** [comment]
+1 passing

**Pull request closed without merging**